core/lib/src/rocket.rs

```diff
@@ -619,7 +619,11 @@ impl Rocket<Ignite> {
             rkt.fairings.handle_liftoff(&rkt).await;
 
             let proto = rkt.config.tls_enabled().then(|| "https").unwrap_or("http");
-            let addr = format!("{}://{}:{}", proto, rkt.config.address, rkt.config.port);
+            let addr = if rkt.config.address.is_ipv4() {
```

Comment:
We could maybe use the existing `Display` impl for `SocketAddr`, which [does the right thing](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d3ae03653bef6043a77034d6aec03aca).

Reply:
Indeed, thank you.
I've just add a commit to the PR in order to use it.